lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -762,3 +785,22 @@ def _tarfile_extract(
     finally:
         if tar_obj is not None:
             tar_obj.close()
+
+
+@cache
+def _get_runtime_from_dir(b_path: bytes) -> object:
+    """Load the meta/runtime.yml from a collection directory."""
+    runtime_path = os.path.join(b_path, b"meta", b"runtime.yml")
+    with suppress(OSError):
+        with open(runtime_path, 'r') as fd:
+            return yaml_load(fd)
```

Comment:
Can we plz start using pathlib for new things internally?
```suggestion
def _get_runtime_from_dir(b_path: bytes) -> object:
    """Load the meta/runtime.yml from a collection directory."""
    runtime_path = Path(to_native(b_path)) / "meta" / "runtime.yml"
    try:
        runtime_yml_txt = runtime_path.read_text(encoding="utf-8")
    except OSError:
        return None

    return yaml_load(runtime_yml_txt)
```

Reply:
Added.